src/h3lib/lib/coordijk.c

```diff
@@ -272,6 +308,96 @@ Direction _unitIjkToDigit(const CoordIJK *ijk) {
     return digit;
 }
 
+/**
+ * Returns non-zero if _upAp7 with the given input could have a signed integer
+ * overflow.
+ *
+ * Assumes ijk is IJK+ coordinates (no negative numbers).
+ */
+H3Error _upAp7Checked(CoordIJK *ijk) {
+    // Doesn't need to be checked because i, j, and k must all be positive
+    int i = ijk->i - ijk->k;
+    int j = ijk->j - ijk->k;
+
+    if (SUM_INT32S_OVERFLOWS(i, i)) {
+        return E_FAILED;
+    }
+    int i2 = i + i;
+    if (SUM_INT32S_OVERFLOWS(i2, i)) {
+        return E_FAILED;
+    }
+    int i3 = i2 + i;
```

Comment:
Maybe it would make sense to make a `MUL_INT32S_OVERFLOWS(i, 3)` check and skip this intermediate addition and check?

Reply:
Is there a reference for a `MUL_INT32S_OVERFLOWS` we can use? I looked at the implementation you pointed to in a conversation last week but that involved floating point math, and I was skeptical there would be a significant difference with the intermediate addition+check approach.